llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -463,6 +471,9 @@ PassBuilder::buildO1FunctionSimplificationPipeline(OptimizationLevel Level,
   LPM1.addPass(LICMPass(PTO.LicmMssaOptCap, PTO.LicmMssaNoAccForPromotionCap,
                         /*AllowSpeculation=*/false));
 
+  if (EnableLoopMetadata)
```

Comment:
Is this the only place where this pass is required? Do we really need to insert several times (check, maybe I miss something)?

Reply:
not every module executes the build01 function simplification pipeline. Therefore we have to add the pass in multiple stages, so that it is always picked up